github/data_source_github_organization_team_sync_groups.go

```diff
@@ -42,7 +42,7 @@ func dataSourceGithubOrganizationTeamSyncGroupsRead(d *schema.ResourceData, meta
 	ctx := context.Background()
 
 	orgName := meta.(*Owner).name
-	options := &github.ListCursorOptions{PerPage: maxPerPage}
+	options := &github.ListIDPGroupsOptions{}
```

Comment:
This is removing the `PerPage` option which is changing the behaviour

perhaps this should be
```suggestion
	options := &github.ListIDPGroupsOptions{
	    ListCursorOptions: github.ListCursorOptions{
	        PerPage: maxPerPage,
        },
	}
```